cmd/crc/cmd/start.go

```diff
@@ -155,3 +157,21 @@ func checkIfNewVersionAvailable(noUpdateCheck bool) {
 	}
 	logging.Debugf("No new version available. The latest version is %s", newVersion)
 }
+
+func handleProxyConfig() (*network.ProxyConfig, error) {
+	httpProxy := crcConfig.GetString(config.HttpProxy.Name)
+	httpsProxy := crcConfig.GetString(config.HttpsProxy.Name)
+	noProxy := crcConfig.GetString(config.NoProxy.Name)
+
+	proxyConfig, err := network.NewProxyConfig(httpProxy, httpsProxy, noProxy)
+
+	if err != nil {
+		return nil, err
+	}
+
+	if proxyConfig.IsEnabled() {
+		logging.Info("Using proxy for the setup")
+		logging.Debugf("HTTP-PROXY: %s, HTTPS-PROXY: %s, NO-PROXY: %s", proxyConfig.HttpProxy(), proxyConfig.HttpsProxy(), proxyConfig.NoProxy())
```

Comment:
how about authentication strings inside the proxy variables?

Reply:
@gbraad this is debug info so only depend on user if want to share.

Reply:
Is the user aware that this might happen. We always should prevent logging username and/or passwords, redact them when possible.

Reply:
@gbraad I do agree on that part so what you suggest, can we just not print it at all then?

Reply:
url.Parse() would allow to print only the bits you want to show.

Reply:
I just wanted to suggest the same... like `Host` is possibly the only important part.

Reply:
... and perhaps this should even be functions on `ProxyConfig`...

Reply:
you want the scheme, the port number. If user and password are set, I think I would print `https://username:xxxx@proxy.example.com` (real username, but replace the password with xxx. This makes it obvious something was set.